examples/e2e/pipelines/batch_inference.py

```diff
@@ -0,0 +1,90 @@
+#  Copyright (c) ZenML GmbH 2023. All Rights Reserved.
+#
+#  Licensed under the Apache License, Version 2.0 (the "License");
+#  you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at:
+#
+#       https://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+#  or implied. See the License for the specific language governing
+#  permissions and limitations under the License.
+
+
+from config import MetaConfig, PipelinesConfig
+from steps import (
+    data_loader,
+    drift_na_count,
+    inference_data_preprocessor,
+    inference_predict,
+    notify_on_failure,
+    notify_on_success,
+)
+from utils.artifacts import find_artifact_id
+
+from zenml import pipeline
+from zenml.integrations.evidently.metrics import EvidentlyMetricConfig
+from zenml.integrations.evidently.steps import evidently_report_step
+from zenml.logger import get_logger
+from zenml.steps.external_artifact import ExternalArtifact
+
+logger = get_logger(__name__)
+
+
+@pipeline(
+    settings={"docker": PipelinesConfig.docker_settings},
+    on_success=notify_on_success,
+    on_failure=notify_on_failure,
```

Comment:
If I remember right, you mentioned this is not working? I believe it should, but if not, you could just do `notify_on_failure` for each step and have `notify_on_success` as a separate pipeline step that runs at the very end of the pipeline

Reply:
Yeah, that's actually better - wrapped `notify_on_success` into step.